arches_component_lab/src/arches_component_lab/widgets/FileListWidget/FileListWidget.vue

```diff
@@ -1,31 +1,19 @@
 <script setup lang="ts">
 import FileListWidgetViewer from "@/arches_component_lab/widgets/FileListWidget/components/FileListWidgetViewer.vue";
-import FileListWidgetEditor from "@/arches_component_lab/widgets/FileListWidget/components/FileListWidgetEditor.vue";
+import FileListWidgetEditor from "@/arches_component_lab/widgets/FileListWidget/components/FileListWidgetEditor/FileListWidgetEditor.vue";
 
 import { EDIT, VIEW } from "@/arches_component_lab/widgets/constants.ts";
 
-import type { CardXNodeXWidget } from "@/arches_component_lab/types.ts";
-import type {
-    WidgetMode,
-    FileReference,
-} from "@/arches_component_lab/widgets/types.ts";
-
-interface FileListCardXNodeXWidgetData extends CardXNodeXWidget {
-    config: CardXNodeXWidget["config"] & {
-        acceptedFiles: string;
-        maxFiles: number;
-        maxFilesize: number;
-        rerender: boolean;
-        label: string;
-    };
-}
+import type { FileListCardXNodeXWidgetData } from "@/arches_component_lab/datatypes/file-list/types.ts";
+import type { WidgetMode } from "@/arches_component_lab/widgets/types.ts";
+import type { NodeData } from "@/arches_component_lab/types.ts";
 
 defineProps<{
     mode: WidgetMode;
     nodeAlias: string;
     graphSlug: string;
     cardXNodeXWidgetData: FileListCardXNodeXWidgetData | undefined;
-    value: FileReference[] | null | undefined;
+    value: NodeData | null | undefined;
```

Comment:
another thing I'm punting -- each datatype should have it's own `types.ts` file with a `MyDatatypeValue` interface that extends NodeData.